pkg/virt-controller/watch/vm.go

```diff
@@ -1285,6 +1294,38 @@ func (c *VMController) updateStatus(vmOrig *virtv1.VirtualMachine, vmi *virtv1.V
 	return err
 }
 
+func (c *VMController) syncReadyConditionFromVMI(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) {
+	vmReadyCond := c.getCondition(vm, virtv1.VirtualMachineReady)
+	vmiReadyCond := controller.NewVirtualMachineInstanceConditionManager().
+		GetCondition(vmi, virtv1.VirtualMachineInstanceConditionType(k8score.PodReady))
+
+	if vmiReadyCond != nil {
+		now := v1.NewTime(time.Now())
+		if vmReadyCond == nil {
+			log.Log.Object(vm).V(4).Info("Adding ready condition")
+			newCond := virtv1.VirtualMachineCondition{
+				Type:               virtv1.VirtualMachineReady,
+				Status:             vmiReadyCond.Status,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             vmiReadyCond.Reason,
+				Message:            vmiReadyCond.Message,
+			}
+			vm.Status.Conditions = append(vm.Status.Conditions, newCond)
+		} else if vmiReadyCond.Status != vmReadyCond.Status {
+			log.Log.Object(vm).V(4).Info("Updating ready condition")
+			vmReadyCond.Status = vmiReadyCond.Status
+			vmReadyCond.LastTransitionTime = now
+			vmReadyCond.Reason = vmiReadyCond.Reason
+			vmReadyCond.Message = vmiReadyCond.Message
+
+		}
+	} else if vmReadyCond != nil {
```

Comment:
This edge case is redundant. I think that it will also improve the readability of this function:

```
if vmiReadyCond == nil {
    c.removeCondition(vm, virtv1.VirtualMachineReady)
    return
}

// now just copy the condition
```

Reply:
That would call the `c.removeCondition` function on every update even when it is not needed. I would prefer to leave the condition there, but I have made it more readable.

Reply:
I don't see a real harm in calling `c.removeCondition` every time (it will do nothing if the condition is nil)...
But it's your code so I don't mind that much.
Thanks for improving the readability :+1: